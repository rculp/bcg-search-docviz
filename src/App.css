.App {
  display: flex;
  flex-direction: column;
  height: 100%;
  min-height: 100%;
  min-height: 100vh;
}

.content {
  flex: 1 1 auto;
  margin: 1em 0;
  width: 100%;
}