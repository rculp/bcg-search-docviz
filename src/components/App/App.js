import React from 'react';
import { hot } from 'react-hot-loader';
import { Provider } from 'react-redux';
import { Route, Switch, Redirect } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';

import { store, history } from 'redux/store';

import HomeContainer from 'containers/Home/Home';
import ResultsContainer from 'containers/Results/Results';

const App = () => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Switch>
        <Route exact path="/" component={HomeContainer} />
        <Route path="/results" component={ResultsContainer} />
        <Redirect from="*" to="/" />
      </Switch>
    </ConnectedRouter>
  </Provider>
);

export default hot(module)(App);
