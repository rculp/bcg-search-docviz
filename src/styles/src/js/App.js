import React, { Component } from 'react';
import { BrowserRouter as Router, Route, NavLink } from "react-router-dom";
import Typography from './Components/Typography';
import Colors from './Components/Colors';
import { Grid, Row, Col } from 'react-flexbox-grid';

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <header className='header'>
            <div className='header__logo'>
              <p className='header__logo-text'>M</p>
            </div>
          </header>
          <Grid fluid>
            <Row>
              <Col xs={12} md={2} lg={1}>
                <nav className='nav-vertical'>
                  <ul className='nav__list'>
                    <li className='nav__list-item'><NavLink to='/typography' className='nav__list-link' activeClassName='nav__list-link--is-active'>Typography</NavLink></li>
                    <li className='nav__list-item'><NavLink to='/colors' className='nav__list-link' activeClassName='nav__list-link--is-active'>Colors</NavLink></li>
                  </ul>
                </nav>
              </Col>
              <Col xs={12} md={10} lg={11}>
                <Route path='/typography' component={Typography} />
                <Route path='/colors' component={Colors} />
              </Col>
            </Row>
          </Grid>
        </div>
      </Router>
    );
  }
}

export default App;
