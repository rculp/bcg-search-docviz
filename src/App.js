import React, { Component } from 'react';

import { BrowserRouter, Route, Link, Switch, Redirect } from 'react-router-dom';

import HomeContainer from 'containers/HomeContainer';
import ResultsContainer from 'containers/ResultsContainer';

import './App.css';

class App extends Component {

  render() {
    return (
      <BrowserRouter>
        <div>
          <Link to="/">Home</Link>
          <Link to="/results">Results</Link>
          <Switch>
            <Route exact path="/" component={HomeContainer}/>
            <Route path="/results" component={ResultsContainer}/>
            <Redirect from="*" to="/"/>
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;

