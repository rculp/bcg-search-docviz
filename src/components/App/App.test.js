import { shallow } from 'enzyme';

import React from 'react';

import App from './App';

describe('App', () => {
  it('renders and matches our snapshot', () => {
    const component = shallow(
      <App />
    );
    expect(component).toMatchSnapshot();
  });
});