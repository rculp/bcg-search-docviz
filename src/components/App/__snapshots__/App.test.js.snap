// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and matches our snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/results"
      />
      <Redirect
        from="*"
        push={false}
        to="/"
      />
    </Switch>
  </Connect(ConnectedRouter)>
</Provider>
`;
